### ISSUE TITLE
AsBindGroup derive macro should link to the AsBindGroup trait documentation

### ISSUE THREAD

**Issue by Lege19** [bevyengine/bevy#16955]
The derive macro documentation is [here](https://docs.rs/bevy/latest/bevy/render/render_resource/derive.AsBindGroup.html).
Currently, the derive attributes (`#[uniform]`, `#[texture_storage]`, etc) are well documented in the [trait](https://docs.rs/bevy/latest/bevy/render/render_resource/trait.AsBindGroup.html) documentation. From the trait perspective, this makes sense, because the vast majority of users will never manually implement this trait, however it would make more sense to put this information in the derive macro documentation and link to it in the trait documentation.
I think that's probably more effort than it's worth though, so instead I'm suggesting to just add a sentence to the macro documentation saying that there is much more detailed information available in the trait documentation (an a link to it).

**Comment by ItsDoot** [comment]
Related: #6195

**Comment by BenjaminBrienen** [comment]
I don't think it is possible to link to an external item like that. Adding the documentation on the re-export breaks the documentation for the derive macro entirely.
